### PR TITLE
Fix to interceptor

### PR DESCRIPTION
**Pull request by simonzuccherato**
Issue #4601

### CODE REVIEW

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/InteractionBlockingInterceptor.java [hunk 1]
```diff
@@ -203,9 +203,25 @@ public Builder addAllowedSpec(String theSpec) {
...
 			String interactionName = theSpec.substring(colonIdx + 1);
-			RestOperationTypeEnum interaction = RestOperationTypeEnum.forCode(interactionName);
-			Validate.notNull(interaction, "Unknown interaction %s in spec %s", interactionName, theSpec);
-			addAllowedInteraction(resourceName, interaction);
+			if (interactionName.equals("search")) {
+				interactionName = "search-type";
+				RestOperationTypeEnum interaction = RestOperationTypeEnum.forCode(interactionName);
+				Validate.notNull(interaction, "Unknown interaction %s in spec %s", interactionName, theSpec);
+				addAllowedInteraction(resourceName, interaction);
+			} else if (interactionName.equals("history")) {
```
Comment (StevenXLi) [inline hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/InteractionBlockingInterceptor.java:211]: The based on the description of this method inputs containing `:` has to be type- and instance-level interactions. You added system level as well, that should not be included.

**Review by StevenXLi — CHANGES_REQUESTED** [review]
You need tests to confirm the changes and make sure the method works as expected. You also need to add a change log.

Also, don't link private gitlab ticket in the description, this is a public repo. Instead should outline what changes you made

**Comment by StevenXLi** [comment]
You should also be creating a new github ticket that outlines the issue you need to fix instead of just creating a branch with the gitlab ticket number and description

File: hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/InteractionBlockingInterceptor.java [hunk 2]
```diff
@@ -203,9 +203,23 @@ public Builder addAllowedSpec(String theSpec) {
...
 			String resourceName = theSpec.substring(0, colonIdx);
 			String interactionName = theSpec.substring(colonIdx + 1);
-			RestOperationTypeEnum interaction = RestOperationTypeEnum.forCode(interactionName);
-			Validate.notNull(interaction, "Unknown interaction %s in spec %s", interactionName, theSpec);
-			addAllowedInteraction(resourceName, interaction);
+			if (interactionName.equals("search")) {
+				interactionName = "search-type";
+				RestOperationTypeEnum interaction = RestOperationTypeEnum.forCode(interactionName);
+				Validate.notNull(interaction, "Unknown interaction %s in spec %s", interactionName, theSpec);
+				addAllowedInteraction(resourceName, interaction);
```
Comment (JasonRoberts-smile) [inline hapi-fhir-server/src/main/java/ca/uhn/fhir/rest/server/interceptor/InteractionBlockingInterceptor.java:210]: You've duplicated this exact block of code four times, differing only in the specific interaction name. That makes it a great candidate to be extracted as a method.

**Review by JasonRoberts-smile — APPROVED** [review]
Approved with a suggestion for improvement

**Pull request merged**